### DISCUSSION TITLE
container create failed under podman 4.6.1 and conmon 2.1.8

### DISCUSSION THREAD

**Discussion by x2byue** [containers/podman#22595] [Q&A]
Does anyone hit this error with podman 4.6.1 and conmon 2.18?

podman run --rm -it test
Error: container create failed (no logs from conmon): conmon bytes "": readObjectStart: expect { or n, but found , error found in #0 byte of ...||..., bigger context ...||...

podman version
Client:       Podman Engine
Version:      4.6.1
API Version:  4.6.1
Go Version:   go1.20.12
Built:        Fri Apr 19 04:09:18 2024
OS/Arch:      linux/amd64

conmon --version
conmon version 2.1.8
commit: 57ed23ee47beaf9a72b28f5666fab095a9ad4a38

**Comment by MajorDallas** [comment]
The issue as existed with those versions is apparently fixed. Some relevant discussion in https://github.com/containers/podman/issues/22274 and similar threads from about the same time.

But for those arriving from <search_engine> with this error while using newer (fixed) versions of podman and conmon, the fix may be as simple as making sure the directory where conmon tries to create the tty socket is accessible to the owner of the conmon process.

In my case on Ubuntu 24.04, the directory is `/var/tmp` and a bit of the company's workstation management software seems to have set the permissions on it to 0755 because reasons. A day later the same software set it back to 0777 while I wasn't looking and suddenly I could get a tty in the container again. I had to dig through `strace` for hours to finally spot this--conmon _really_ needs better error reporting.
